docs/software/cloud.md

```diff
@@ -646,6 +647,126 @@ Official forum: <https://forum.gitea.com/>
+
+    - Port: 3000
+    - Protocol: TCP
+
+    If an external access is used, HTTPS is strongly recommended to increase your system security. You can get a free certificate e.g. via [dietpi-letsencrypt](../dietpi_tools.md#dietpi-letsencrypt){:class="nospellcheck"}.
+
+    2 - Without port forwarding, you can setup a Tunnel or Reverse Proxy with services like [Cloudflare Tunnel]([https://www.cloudflare.com/en-ca/products/tunnel/](https://developers.cloudflare.com/cloudflare-one/connections/connect-networks/get-started/create-remote-tunnel/))
+
+=== "Fail2Ban integration"
+
+    Using Fail2Ban your can block IP addresses after failed login attempts. This hardens your system against e.g. brute-force attacks.
+
+    === "When using journal logging (default)"
+
+        By default Forgejo logs to the systemd journal (see "View logs" tab), in which case Fail2Ban protection can be enabled with the following steps:
+
+        Create a new filter `/etc/fail2ban/filter.d/forgejo.conf`:
+
+        ```ini
+        [Definition]
+        journalmatch = _SYSTEMD_UNIT=forgejo.service
+        failregex = Failed authentication attempt for \x1b\[1m.+\x1b\[0m from \x1b\[1m<HOST>:\d+\x1b\[0m:
+        ```
+
+        Create a new jail `/etc/fail2ban/jail.d/forgejo.conf`:
+
+        ```ini
+        [forgejo]
+        enabled = true
+        backend = systemd
+        ```
+
+    === "When using file logging"
+
+        When file logging is enabled (see "View logs" tab), Fail2Ban protection can be enabled with the following steps:
+
+        Create a new filter `/etc/fail2ban/filter.d/forgejo.conf`:
+
+        ```ini
+        [Definition]
+        failregex = Failed authentication attempt for .+ from <HOST>:\d+:
+        ```
+
+        Create a new jail `/etc/fail2ban/jail.d/forgejo.conf`:
+
+        ```ini
+        [forgejo]
+        enabled = true
+        backend = auto
+        logpath = /var/log/forgejo/forgejo.log
+        ```
+
+    You can specify other properties like `maxretry` or `bantime` to overwrite the defaults in `/etc/fail2ban/jail.conf` or `/etc/fail2ban/jail.local` if present. When done:
+
+    - Restart Fail2Ban: `systemctl restart fail2ban`
+    - Try to login to the Forgejo web interface with a wrong password.
+    - Check whether the failed login has been detected: `fail2ban-client status forgejo`
+    - When you further try to login `maxretry` times, your IP should be banned for `bantime` seconds, so that neither the Forgejo web interface, nor SSH or any other network application will respond to requests from your client. When Fail2Ban was installed via `dietpi-software`, by default `route`/`blackhole` blocking is used, so that `ip r` on the server should show a `blackhole` route for your client's IP.
+    - See also:
+        - [Fail2Ban](system_security.md#fail2ban)
+        - <https://docs.forgejo.com/administration/fail2ban-setup>
```

Comment:
These docs do not exist. I suggest to remove the Fail2Ban section, until someone has time to test and verify that the same rules used for Gitea work for Forgejo as well.